### DISCUSSION TITLE
analyzer dependency conflict

### DISCUSSION THREAD

**Discussion by siluvana** [vitusortner/floor#665] [Q&A] [answered]
Hi. I was doing some changes in my entities database, and when I try to run _flutter packages pub run build_runner build_ I get a warning from floor_generator about analyzer which ends up in a severe error and can't generate the database file.

That's the detail:

**[WARNING] floor_generator:floor_generator on test/widget_test.dart:
Your current `analyzer` version may not fully support your current SDK version.

Analyzer language version: 2.16.0
SDK language version: 2.17.0

Please update to the latest `analyzer` version (4.1.0) by running
`flutter packages upgrade`.

If you are not getting the latest version by running the above command, you
can try adding a constraint like the following to your pubspec to start
diagnosing why you can't get the latest version:

dev_dependencies:
  analyzer: ^4.1.0**

Then it shows this message:

**[SEVERE] floor_generator:floor_generator on test/widget_test.dart:

Invalid argument(s): Missing library: asset:myproject/lib/BloodyMakers/CategoryMaker.dart
Available libraries: [dart:typed_data, dart:convert, dart:math, dart:core, dart:_internal, dart:collection, dart:async, dart:isolate, dart:developer, dart:ffi, dart:_js_embedded_names, dart:_js_names, dart:_recipe_syntax, dart:_rti, dart:_foreign_helper, dart:_js_helper, dart:_interceptors, dart:_native_typed_data, dart:web_gl, dart:js_util, dart:_metadata, dart:html_common, dart:indexed_db, dart:svg, dart:web_audio, dart:html, dart:io, dart:_http, dart:js, dart:_js, dart:mirrors, dart:nativewrappers, dart:cli, dart:_late_helper, dart:_dart2js_runtime_metrics, dart:_js_primitives, dart:_async_await_error_codes, dart:_js_annotations, dart:_spirv, dart:ui]
#0      LinkedElementFactory.createLibraryElementForReading (package:analyzer/src/summary2/linked_element_factory.dart:74:7)
#1      LinkedElementFactory.elementOfReference (package:analyzer/src/summary2/linked_element_factory.dart:128:14)
#2      LinkedElementFactory.libraryOfUri (package:analyzer/src/summary2/linked_element_factory.dart:177:12)
#3      ElementBuilder._selectLibrary (package:analyzer/src/summary2/element_builder.dart:1018:37)
#4      ElementBuilder.visitImportDirective (package:analyzer/src/summary2/element_builder.dart:598:31)
#5      ImportDirectiveImpl.accept (package:analyzer/src/dart/ast/ast.dart:5953:50)
#6      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7758:20)
#7      ElementBuilder.buildLibraryElementChildren (package:analyzer/src/summary2/element_builder.dart:60:21)
#8      LibraryBuilder.buildElements (package:analyzer/src/summary2/library_builder.dart:109:24)
#9      Linker._computeLibraryScopes (package:analyzer/src/summary2/link.dart:109:15)
#10     Linker._buildOutlines (package:analyzer/src/summary2/link.dart:88:5)
#11     Linker.link (package:analyzer/src/summary2/link.dart:74:5)
#12     link (package:analyzer/src/summary2/link.dart:33:10)
#13     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:176:30)
#14     SetMixin.forEach (dart:collection/set.dart:142:30)
#15     LibraryContext.load2.loadBundle (package:analyzer/src/dart/analysis/library_context.dart:113:32)
#16     LibraryContext.load2.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:206:7)
#17     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#18     LibraryContext.load2 (package:analyzer/src/dart/analysis/library_context.dart:204:12)
#19     AnalysisDriver._computeUnitElement.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1574:22)
#20     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#21     AnalysisDriver._computeUnitElement (package:analyzer/src/dart/analysis/driver.dart:1572:20)
#22     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1126:35)
#23     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2140:24)
<asynchronous suspension>

[INFO] 16.0s elapsed, 73/79 actions completed.**

and keeps showing _73/79 actions completed_ infinitelly.

----
If I put analyzer 4.1.0 as dev_dependencies, I have unresolved dependencies conflict.

`Because floor_generator 1.2.0 depends on analyzer ^2.2.0 and no versions of floor_generator match >1.2.0 <2.0.0, floor_generator ^1.2.0 requires analyzer ^2.2.0.
So, because myproject depends on both floor_generator ^1.2.0 and analyzer ^4.1.0, version solving failed.`

Is there a solution for this?

Thanks!
Silvana.

**Comment by siluvana** [comment]
I've just found that's there is a PR waiting for this dependency update. I hope it'll solve it.

**Reply by Amir-P** [reply]
You can use https://github.com/Amir-P/flat until they fix it on Floor. @siluvana

**Reply by siluvana** [reply]
Thanks, Amir!

**Comment by dkaera** [comment, marked as answer]
Pushed update [here](https://github.com/vitusortner/floor/pull/666) but need some help to solve GitHub CI Action issues.

**Reply by dkaera** [reply]
Update will be soon, thanks @ThomasMiddel 👍

**Reply by dkaera** [reply]
@siluvana please use the latest Floor version.

**Reply by siluvana** [reply]
Thanks a lot for your answer. Resuming the project after a while and is kinda working now with the new version.
